### PR TITLE
enhancement to combine alike filters

### PR DESCRIPTION
**Pull request by mstumpfx**
combine filters of same type (same monitorType, negation, and dimensions)

### CODE REVIEW

**Comment by keitwb** [comment]
@mstumpfx when do you think you will be able to get to this?

**Comment by mstumpfx** [comment]
@keitwb I'll get to it this week sometime

File: tests/basic/filter_test.py [hunk 1]
```diff
@@ -86,6 +86,46 @@ def test_combined_filter_with_monitor_type():
...
+def test_overlapping_filter_with_monitor_type2():
+    with run_agent(
+        """
+monitors:
+  - type: collectd/memory
+  - type: collectd/uptime
+metricsToExclude:
+  - metricName: uptime
+    monitorType: collectd/uptime
+  - metricName: uptime
```
Comment (keitwb) [inline tests/basic/filter_test.py]: What does this mean for two contradictory filters?

Comment (mstumpfx) [reply tests/basic/filter_test.py]: test_overlapping_filter_with_monitor_type and test_overlapping_filter_with_monitor_type2
show that order in which filters are defined does not matter, and that exclusion of metrics will always take precedence over "negated:true"

File: tests/basic/filter_test.py [hunk 2]
```diff
@@ -60,6 +60,72 @@ def test_negated_filter_with_monitor_type():
...
+    negated: true
+"""
+    ) as [backend, _, _]:
+        assert wait_for(lambda: has_datapoint_with_metric_name(backend, "memory.used"))
+        assert wait_for(lambda: has_datapoint_with_metric_name(backend, "memory.free"))
+        assert ensure_always(lambda: not has_datapoint_with_metric_name(backend, "memory.cached"), 10)
+        assert ensure_always(lambda: not has_datapoint_with_metric_name(backend, "uptime"), 5)
+
+
+# Test overlapping filters with different negation. Blacklist is favored
```
Comment (keitwb) [inline tests/basic/filter_test.py]: Put this as a docstring it will be much clearer.

**Pull request merged**